Libraries/LibTLS/TLSv12.h

```diff
@@ -0,0 +1,400 @@
+
+    const StringView& alpn() const { return m_context.negotiated_alpn; }
+    void add_alpn(const StringView& alpn);
+    bool has_alpn(const StringView& alpn) const;
+
+    bool cipher_supported(CipherSuite suite) const
+    {
+        return suite == CipherSuite::RSA_WITH_AES_128_CBC_SHA256 || suite == CipherSuite::RSA_WITH_AES_256_CBC_SHA256;
+    }
+
+    bool version_supported(Version v) const
+    {
+        return v == Version::V12;
+    }
+
+    Optional<ByteBuffer> read();
+    ByteBuffer read(size_t max_size);
+
+    bool write(const ByteBuffer& buffer);
+    void alert(bool critical, u8 code);
+
+    Function<void(TLSv12&)> on_tls_ready_to_read;
+    Function<void(TLSv12&)> on_tls_ready_to_write;
+    Function<void(AlertDescription)> on_tls_error;
+    Function<void()> on_tls_connected;
+    Function<void()> on_tls_finished;
+
+private:
+    bool common_connect(const struct sockaddr*, socklen_t) override;
+
+    void consume(const ByteBuffer& record);
+
+    ByteBuffer hmac_message(const ByteBuffer& buf, const Optional<ByteBuffer> buf2, size_t mac_length, bool local = false);
+
+    void update_packet(ByteBuffer& packet);
+    void update_hash(const ByteBuffer& in);
+
+    void write_packet(ByteBuffer& packet);
+
+    ByteBuffer build_client_key_exchange();
+    ByteBuffer build_server_key_exchange();
+
+    ByteBuffer build_hello();
+    ByteBuffer build_finished();
+    ByteBuffer build_certificate();
+    ByteBuffer build_done();
+    ByteBuffer build_alert(bool critical, u8 code);
+    ByteBuffer build_change_cipher_spec();
+    ByteBuffer build_verify_request();
+    void build_random(PacketBuilder&);
+
+    bool flush();
+
+    ssize_t handle_hello(const ByteBuffer& buffer, size_t&);
+    ssize_t handle_finished(const ByteBuffer& buffer, size_t&);
+    ssize_t handle_certificate(const ByteBuffer& buffer);
+    ssize_t handle_server_key_exchange(const ByteBuffer& buffer);
+    ssize_t handle_server_hello_done(const ByteBuffer& buffer);
+    ssize_t handle_verify(const ByteBuffer& buffer);
+    ssize_t handle_payload(const ByteBuffer& buffer);
+    ssize_t handle_message(const ByteBuffer& buffer);
+    ssize_t handle_random(const ByteBuffer& buffer);
+
+    size_t asn1_length(const ByteBuffer& buffer, size_t* octets);
+
+    void pseudorandom_function(ByteBuffer& output, const ByteBuffer& secret, const u8* label, size_t label_length, const ByteBuffer& seed, const ByteBuffer& seed_b);
+
+    size_t key_length() const { return m_aes_local ? m_aes_local->cipher().key().length() : 16; } // FIXME: generalize
+    size_t mac_length() const { return Crypto::Authentication::HMAC<Crypto::Hash::SHA256>::DigestSize; } // FIXME: generalize
+
+    bool expand_key();
+
+    bool compute_master_secret(size_t length);
+
+    Context m_context;
+    RefPtr<Core::Notifier> m_notifier;
+
+    OwnPtr<Crypto::Cipher::AESCipher::CBCMode> m_aes_local { nullptr };
+    OwnPtr<Crypto::Cipher::AESCipher::CBCMode> m_aes_remote { nullptr };
```

Comment:
No need to explicitly initialize smart pointers; they are null by default.

Reply:
You will keep telling me this, and I will keep doing it because my brain goes "these start null"